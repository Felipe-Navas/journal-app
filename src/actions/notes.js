import { db } from "../firebase/firebase-config";
import { types } from '../types/types'

export const startNewNote = () => {
  return async (dispatch, getState) => {
    const uid = getState().auth.uid

    const newNote = {
      body: '',
      date: new Date().getTime(),
      title: '',
    }

    const doc = await db.collection(`${uid}/journal/notes`).add(newNote)
    dispatch(activeNote(doc.id, newNote))
  }
}

export const activeNote = (id, note) => ({
  type: types.notesActive,
  payload: {
    id,
    ...note,
  },
})

export const setNotes = (notes) => ({
  type: types.notesLoad,
  payload: notes,
})
