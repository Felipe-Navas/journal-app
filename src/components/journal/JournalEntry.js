import React from 'react';

export const JournalEntry = () => {
  return (
    <div className="journal__entry pointer">
      <div
        className="journal__entry-picture"
        style={{
          backgroundSize: 'cover',
          backgroundImage:
            'url(https://images.unsplash.com/photo-1518791841217-8f162f1e1131?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=60)',
        }}
      ></div>
      <div className="journal__entry-body">
        <p className="journal__entry-title">A simple cat</p>
        <p className="journal__entry-content">
          Lorem ipsum dolor, sit amet consectetur adipisicing elit.
        </p>
      </div>
      <div className="journal__entry-date-box">
        <span>Monday</span>
        <h4>25</h4>
      </div>
    </div>
  );
};
