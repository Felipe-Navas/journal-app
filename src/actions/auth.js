import Swal from 'sweetalert2';
import { types } from '../types/types';
import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import { finishLoading, startLoading } from './ui';
import { noteLogout } from './notes'

/**
 *
 * @param {String} uid
 * @param {String} displayName
 * @returns Object
 */
export const login = (uid, displayName) => {
  return {
    type: types.login,
    payload: {
      uid,
      displayName,
    },
  };
};

/**
 *
 * @param {String} email
 * @param {String} password
 * @returns
 */
export const startLoginEmailPassword = (email, password) => {
  return (dispatch) => {
    dispatch(startLoading());

    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
        dispatch(finishLoading());
      })
      .catch((err) => {
        console.log(err);
        dispatch(finishLoading());
        Swal.fire('Error', err.message, 'error');
      });
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
      });
  };
};

export const startRegisterWithEmailPasswordName = (email, password, name) => {
  return (dispatch) => {
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({ displayName: name });
        dispatch(login(user.uid, user.displayName));
      })
      .catch((err) => {
        console.log(err);
        Swal.fire('Error', err.message, 'error');
      });
  };
};

export const startLogout = () => {
  return async (dispatch) => {
    await firebase.auth().signOut();
    dispatch(logout());
    dispatch(noteLogout())
  };
};

export const logout = () => {
  return {
    type: types.logout,
  };
};
