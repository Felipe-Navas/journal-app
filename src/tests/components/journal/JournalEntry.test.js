import React from 'react'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { JournalEntry } from '../../../components/journal/JournalEntry'
import { activeNote } from '../../../actions/notes'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initState = {}

let store = mockStore(initState)

store.dispatch = jest.fn()

const note = {
  id: 1,
  date: '2020-01-01',
  title: 'Title',
  body: 'Body',
  url: 'https://www.google.com',
}

const wrapper = mount(
  <Provider store={store}>
    <JournalEntry {...note} />
  </Provider>
)

describe('Testing the <JournalEntry /> component', () => {
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should call the activeNote', () => {
    wrapper.find('.journal__entry').simulate('click')
    expect(store.dispatch).toHaveBeenLastCalledWith(activeNote(1, { ...note }))
  })
})
