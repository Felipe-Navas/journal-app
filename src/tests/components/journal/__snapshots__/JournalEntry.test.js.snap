// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <JournalEntry /> component should render correctly 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://www.google.com)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      Title
    </p>
    <p
      className="journal__entry-content"
    >
      Body
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Wednesday
    </span>
    <h4>
      1st
    </h4>
  </div>
</div>
`;
